### PR TITLE
Changes of Student and Teacher Validation 

### PR DESCRIPTION
**Pull request by Himanshu0k**
Student aur teacher dono m validation file m allowed fields ka code apply kar diya h. Ek baar review karke bata do bhaiya further changes ......

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "main": "app.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "nodemon ./src/app.js"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "dotenv": "^16.4.5",
```
Comment (GargPermil) [inline package.json:16]: add other packages like lint, husky.

File: src/controllers/login/login.controller.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        // Check if the username matches the one from .env
+        if (username !== ENV_USERNAME) {
+            return res.status(401).json({ status: 401, error: 'Invalid username' });
+        }
+
+        // Check if the password matches the one from .env
+        if (password !== ENV_PASSWORD) {
+            return res.status(401).json({ status: 401, error: 'Invalid password' });
+        }
```
Comment (GargPermil) [inline src/controllers/login/login.controller.js]: validation should be in validation file.

File: src/controllers/students/student.controller.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        // Check if student with this auto-generated ID already exists
+        if (studentModel.studentExists(id)) {
+            return res.status(400).json({ status: 400, error: 'Student with this ID already exists' });
+        }
+    
+        // Trim the input fields
+        name = name.trim();
+        address = address.trim();
+        gender = gender.trim();
+        course_name = course_name.trim();
```
Comment (GargPermil) [inline src/controllers/students/student.controller.js:23]: do we need trim here ??

File: src/controllers/students/student.controller.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        // Add the student to the model
+        studentModel.addStudent(student);
+    
+        // Respond with success
+        res.status(201).json({ status: 201, message: 'Added student details successfully', student });
+    },
+    
+
+    fetchAllStudents: (req, res) => {
+        console.log("kuch bhi")
```
Comment (GargPermil) [inline src/controllers/students/student.controller.js]: remove this consoles

File: src/controllers/students/student.controller.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            // Restrict changes to "id"
+            if (key === 'id') {
+                const errorMessage = 'You cannot change the "id".';
+                console.error(errorMessage);
+                return res.status(400).json({ status: 400, error: errorMessage });
+            }
+    
+            // Validate and update name if present
+            if (key === 'name') {
+                if (!updatedData[key].trim() || updatedData[key].trim().length < 3 || /\d/.test(updatedData[key]) || /[^a-zA-Z\s]/.test(updatedData[key])) {
```
Comment (GargPermil) [inline src/controllers/students/student.controller.js]: do we dont have this validation in validation file ?

File: Backend_StudyIn
Comment (GargPermil) [inline Backend_StudyIn:1]: what is this file ?

File: src/controllers/students/student.controller.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+        // Validate and update only the fields that are present in the request body
+        for (const key of Object.keys(updatedData)) {
+            // Restrict changes to "id"
+            if (key === 'id') {
+                return errorResponse(res, 'You cannot change the id');
+            }
+
+            // Validate and update each allowed field
+            if (allowedFields.includes(key)) {
```
Comment (GargPermil) [inline src/controllers/students/student.controller.js]: can we add this validation in validation file ?

File: src/controllers/students/student.controller.js (hunk 3 above)
Comment (Himanshu0k) [reply src/controllers/students/student.controller.js:23]: the trim here will remove all the extra spaces entered by the user

**Pull request merged**